File: Medium/2025-11-12-2654-Minimum-Number-of-Operations-to-Make-All-Array-Elements-Equal-to-1/solution.java
import java.util.Arrays;

class Solution {
    public int minOperations(int[] nums) {
        if (Arrays.stream(nums).allMatch(num -> num == 1)) {
            return 0;
        }
        
        int gcd = nums[0];
        for (int num : nums) {
            gcd = gcd(gcd, num);
            if (gcd == 1) {
                break;
            }
        }
        
        if (gcd > 1) {
            return -1;
        }
        
        int operations = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            if (gcd(nums[i], nums[i + 1]) == 1) {
                operations++;
            }
        }
        
        return operations + (nums.length - 1);
    }
    
    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

File: Medium/2025-11-12-2654-Minimum-Number-of-Operations-to-Make-All-Array-Elements-Equal-to-1/solution.js
function minOperations(nums) {
    const gcd = (a, b) => b === 0 ? a : gcd(b, a % b);
    
    let overallGCD = nums[0];
    for (let num of nums) {
        overallGCD = gcd(overallGCD, num);
    }
    
    if (overallGCD !== 1) return -1;

    let operations = 0;
    for (let i = 0; i < nums.length - 1; i++) {
        if (gcd(nums[i], nums[i + 1]) === 1) {
            operations += 1;
        }
    }
    
    return operations + (nums.length - 1);
}